thunder/tests/test_examine_memory.py

```diff
@@ -71,19 +71,12 @@ def bar(a, b):  # [4] [2,2]
     bw_traces = thunder.last_backward_traces(cbar)
     bw_extrace = bw_traces[-1]
     max_mem_bw = get_alloc_memory(bw_extrace)
-
-    # NOTE: nvFuser is able to avoid the allocation of result2 and produce it as alias to result1.
-    if executor.name == "nvfuser":
-        assert max_mem_bw[0] == 128
-    else:
-        assert max_mem_bw[0] == 144
+    # nvFuser should be able to avoid the allocation of result2 and produce it as alias to result1, reducing the allocated memory to 128. However,
+    # due to a limitation of `get_alloc_memory` (https://github.com/Lightning-AI/lightning-thunder/blob/6dfe7e939a19d1ef5ab259de8709a79f0104fa42/thunder/examine/memory_caculation.py#L123-L125), this saved memory is not taken into consideration.
+    assert max_mem_bw[0] == 144
```

Comment:
As I said in the comment above, I made this test change because, with this PR, 
1. `result2` (a reshape) is done inside the nvFusion instead, and
2. `get_alloc_memory` overestimates nvFuser's memory usage, assuming that nvFuser allocates memory for both `result1` and `result2`. 

CSE does still kicks in as expected. 

With this PR and the following patch
```
diff --git a/thunder/tests/test_examine_memory.py b/thunder/tests/test_examine_memory.py
index da92e966..b8537c4d 100644
--- a/thunder/tests/test_examine_memory.py
+++ b/thunder/tests/test_examine_memory.py
@@ -70,6 +70,7 @@ def test_view_ops(executor, device: str, dtype: dtypes.dtype):
     assert sum(max_mem_fwd[1].values()) == get_return_memory(fwd_extrace.bound_symbols[-1])  # 144
     bw_traces = thunder.last_backward_traces(cbar)
     bw_extrace = bw_traces[-1]
+    print(bw_extrace)
     max_mem_bw = get_alloc_memory(bw_extrace)
     # nvFuser should be able to avoid the allocation of result2 and produce it as alias to result1, reducing the allocated memory to 128. However,
     # due to a limitation of `get_alloc_memory` (https://github.com/Lightning-AI/lightning-thunder/blob/6dfe7e939a19d1ef5ab259de8709a79f0104fa42/thunder/examine/memory_caculation.py#L123-L125), this saved memory is not taken into consideration.
```

```
$ pytest thunder/tests/test_examine_memory.py -k test_view_ops -s
# Constructed by Delete Last Used (took 0 milliseconds)
import torch
from thunder.executors.torchex import no_autocast

@torch.no_grad()
@no_autocast
def backward_fn(saved_for_backward, cotangents):
  # cotangents: "Collection"
  t2, t3, = cotangents
  clear_mutable_collection(cotangents)
  del cotangents
  [t58, t60] = nvFusion0(t2, t3)
    # t53 = prims.sum(t3, (0, 1))  # t53: "cuda:0 f32[4]"
    # t54 = prims.broadcast_in_dim(t53, [1, 1, 4], [2])  # t54: "cuda:0 f32[1, 1, 4]"
    # t55 = prims.add(t2, t54)  # t55: "cuda:0 f32[1, 1, 4]"
    # t56 = prims.sum(t55, (0, 1))  # t56: "cuda:0 f32[4]"
    # t57 = prims.broadcast_in_dim(t56, [1, 4], [1])  # t57: "cuda:0 f32[1, 4]"
    # t58 = prims.sum(t57, (0,))  # t58: "cuda:0 f32[4]"
    # t60 = prims.reshape(t58, (2, 2))  # t60: "cuda:0 f32[2, 2]"
  del t2, t3
  return (t58, t60)
```

There are three `sum`s, meaning CSE indeed kicks in.